### PR TITLE
patch for search issues by key using JIRA issuePicker rest API.

### PR DESCRIPTION
**Pull request by ankushmiddha**
Added search_for_issue_by_key method in forked JIRA class

### CODE REVIEW

**Comment by ankushmiddha** [comment]
@duttly @sirsgriffin  @tgrrtt 

Added method in forked version JIRA class. Follow up PR of loggly/fronted (https://github.com/loggly/frontend/pull/6836)
Please review.

File: jira/client.py [hunk 1]
```diff
@@ -1737,6 +1737,25 @@ def search_issues(self, jql_str, startAt=0, maxResults=50, validate_query=True,
...
+        :param search_key: input from user for the issue to be found
+        :return: JSON response will be returned if issues found; otherwise returns, msg:No match found.
+
+        In constatn JIRA_API_ISSUE_SEARCH, showSubTasks=true and showSubTaskParent=true will get sub & parent
+        tasks of searched result currentJQL=' (leaving it blank) will search in the whole corpus, not just
+        Recently-viewed issues. which is the default. Doc is at
+        https://docs.atlassian.com/jira/REST/cloud/#api/2/issue-getIssuePickerResource
+        '''
+        JIRA_API_ISSUE_SEARCH = "/rest/api/2/issue/picker" \
+                                "?showSubTasks=true&showSubTaskParent=true&currentJQL=&query="
```
Comment (tgrrtt) [inline jira/client.py]: All of these querystrings should be parameterized.

Comment (tgrrtt) [inline jira/client.py:1754]: All of these querystrings should be parameterized.

**Review by tgrrtt — CHANGES_REQUESTED** [review]

File: jira/client.py [hunk 2]
```diff
@@ -1737,6 +1737,25 @@ def search_issues(self, jql_str, startAt=0, maxResults=50, validate_query=True,
...
+        search_url = "{}{}{}".format(self._options["server"], JIRA_API_ISSUE_SEARCH, search_key)
+        search_result = self._session.get(search_url)
```
Comment (sirsgriffin) [inline jira/client.py:1756]: Shouldn't this attempt adhere to the current convention in this module of using self._fetch_pages()?

Comment (tgrrtt) [reply jira/client.py:1756]: I think the other `search` functions take pagination params.

https://docs.atlassian.com/jira/REST/cloud/#api/2/issue-getIssuePickerResource
Doesn't seem to be paginated

Comment (tgrrtt) [reply jira/client.py:1756]: As it's not an actual "search"

Comment (duttly) [reply jira/client.py:1756]: Correct, no pagination. It returns N results (divided in 2 sections), we have no control of N or the split.

File: jira/client.py (hunk 1 above)
Comment (duttly) [reply jira/client.py:1754]: @tgrrtt There is no current plan to tweak the parameters, or expose them in some form. Keeping it simple for now.

**Pull request merged**